src/SDK/Trace/TracerProviderBuilder.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Trace;
+
+use OpenTelemetry\SDK\Common\Future\CancellationInterface;
+use OpenTelemetry\SDK\Common\Util\ShutdownHandler;
+use OpenTelemetry\SDK\Resource\ResourceInfo;
+use OpenTelemetry\SDK\Resource\ResourceInfoFactory;
+use OpenTelemetry\SDK\Trace\Sampler\AlwaysOnSampler;
+use OpenTelemetry\SDK\Trace\Sampler\ParentBased;
+
+class TracerProviderBuilder
+{
+    // @var array<SpanProcessorInterface>
+    private ?array $spanProcessors = [];
+    private ?ResourceInfo $resource = null;
+    private ?SamplerInterface $sampler = null;
+    private bool $autoShutdown = false;
+
+    public function addSpanProcessor(SpanProcessorInterface $spanProcessor): self
+    {
+        $this->spanProcessors[] = $spanProcessor;
+
+        return $this;
+    }
+
+    public function setResource(ResourceInfo $resource): self
```

Comment:
IMO this belongs in the `SdkBuilder`.

Reply:
This one, I'm not sure about. The SdkBuilder currently expects to be given a tracer & meter provider (with resources already provided, since it's a constructor arg).
The only way I can think of this working is if the SdkBuilder actually builds the tracer/meter providers (which is not what java does, and that was my inspiration: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/all/src/main/java/io/opentelemetry/sdk/OpenTelemetrySdkBuilder.java )

Reply:
Java is a good reference lang to be following for otel semantics.